frontend/src/scenes/insights/insightVizDataLogic.ts

```diff
@@ -537,13 +548,76 @@ const handleQuerySourceUpdateSideEffects = (
     // if mixed, clear breakdown and trends filter
     if (
         kind === NodeKind.TrendsQuery &&
-        (mergedUpdate as TrendsQuery).series?.length >= 0 &&
-        (mergedUpdate as TrendsQuery).series.some((series) => isDataWarehouseNode(series)) &&
-        (mergedUpdate as TrendsQuery).series.some((series) => isActionsNode(series) || isEventsNode(series))
+        trendsMergedUpdate.series?.length >= 0 &&
+        trendsMergedUpdate.series.some((series) => isDataWarehouseNode(series)) &&
+        trendsMergedUpdate.series.some((series) => isActionsNode(series) || isEventsNode(series))
     ) {
         mergedUpdate['breakdownFilter'] = null
         mergedUpdate['properties'] = []
     }
 
+    // Don't allow minutes on anything other than Trends
+    if (
+        currentState.kind == NodeKind.TrendsQuery &&
+        kind !== NodeKind.TrendsQuery &&
+        (('interval' in mergedUpdate && mergedUpdate?.interval) || interval) == 'minute'
+    ) {
+        trendsMergedUpdate.interval = 'hour'
+    }
+
+    // Reset dateRange to the dateMapping default
+    if (kind == NodeKind.TrendsQuery && trendsMergedUpdate?.interval == 'minute' && interval !== 'minute') {
+        const { date_from, date_to } = { ...currentState.dateRange, ...update.dateRange }
+
+        // Change the date range to be the last 3 hours if it was longer
+        const dateMapping = realTimeDateMapping[1]
```

Comment:
It's not obvious or guaranteed this will be 3 hours. `.find()` would be more resilient